pkg/cmd/chat/user/user.go

```diff
@@ -296,3 +305,310 @@ func revokeCmd() *cobra.Command {
+		`),
+		Example: heredoc.Doc(`
+			# Deactivate the user 'joe'
+			$ stream-cli chat deactivate-user --user-id joe --mark-messages-deleted true
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			targetID, _ := cmd.Flags().GetString("user-id")
+			markDeleted, _ := cmd.Flags().GetBool("mark-messages-deleted")
+
+			deactivateOpts := []stream.DeactivateUserOptions{}
+			if markDeleted {
+				deactivateOpts = append(deactivateOpts, stream.DeactivateUserWithMarkMessagesDeleted())
+			}
+
+			_, err = c.DeactivateUser(cmd.Context(), targetID, deactivateOpts...)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully deactivated user")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("user-id", "u", "", "[required] ID of the user to deactivate")
+	fl.Bool("mark-messages-deleted", false, "[optional] Mark all messages from the user as deleted")
+	_ = cmd.MarkFlagRequired("user-id")
+
+	return cmd
+}
+
+func muteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "mute-user --target-user-id [user-id] --muted-by-id [user-id] --expiration [minutes]",
+		Short: "Mute a user",
+		Long: heredoc.Doc(`
+			Any user is allowed to mute another user. Mutes are stored at the user
+			level and returned with the rest of the user information when connectUser is called.
+			A user will be muted until the user is unmuted or the mute is expired.
+		`),
+		Example: heredoc.Doc(`
+			# Mute the user 'joe' for 5 minutes
+			$ stream-cli chat mute-user --target-user-id joe --muted-by-id admin --expiration 5
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			targetID, _ := cmd.Flags().GetString("target-user-id")
+			mutedByID, _ := cmd.Flags().GetString("muted-by-id")
+			expiration, _ := cmd.Flags().GetInt("expiration")
+
+			muteOpts := []stream.MuteOption{}
+			if expiration > 0 {
+				muteOpts = append(muteOpts, stream.MuteWithExpiration(expiration))
+			}
+
+			_, err = c.MuteUser(cmd.Context(), targetID, mutedByID, muteOpts...)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully muted user")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("target-user-id", "t", "", "[required] ID of the user to mute")
+	fl.StringP("muted-by-id", "b", "", "[required] ID of the user who muted the user")
+	fl.IntP("expiration", "e", 0, "[optional] Number of minutes until the mute expires")
+	_ = cmd.MarkFlagRequired("target-user-id")
+	_ = cmd.MarkFlagRequired("muted-by-id")
+
+	return cmd
+}
+
+func unmuteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unmute-user --target-user-id [user-id] --unmuted-by-id [user-id]",
+		Short: "Unmute a user",
+		Long: heredoc.Doc(`
+			Any user is allowed to mute another user. Mutes are stored at the user
```

Comment:
maybe just say that this unmute a muted user?

Reply:
✅